src/cata_tiles.h

```diff
@@ -325,35 +329,59 @@ class cata_tiles
         void draw_minimap( const point &dest, const tripoint &center, int width, int height );
 
     protected:
+        struct tile_render_info {
+            const tripoint pos{};
+            // accumulator for 3d tallness of sprites rendered here so far;
+            mutable int height_3d = 0;
```

Comment:
At the same time, if I make the argument `tile_render_info &`, all fields could be mutated, which is even worse. How would you suggest to solve this?  I don't want to add more parameters to the `draw_...` methods, especially since they all have the same signature.

So far I'm thinking of maybe adding a comment that clearly explains the `mutable` usage there. Would that suffice?

Reply:
The best I can think of is to make every value that isn't mutated const, and use constructors everywhere to construct `tile_render_info`. This could make construction more verbose though, but I think it's better than making the method signature confusing.

Reply:
Thanks, I'll try that. I also came up with another idea: to split `tile_render_info` into two classes, first with immutable and second with mutable data, but that might require a bit more work.